internal/npm_install/generate_pnp_build_files.ts

```
@@ -0,0 +1,44 @@
+import * as fs from "fs";
+import * as path from "path";
+
+const workspacePath = process.argv.slice(2)[0];
+
+function mkdirp(p: string) {
+  if (!fs.existsSync(p)) {
+    mkdirp(path.dirname(p));
+    fs.mkdirSync(p);
+  }
+}
+
+function writeFileSync(p: string, content: string) {
+  mkdirp(path.dirname(p));
+  fs.writeFileSync(p, content);
+}
+
+function createBuildFile(pkg: string) {
+  const contents = `package(default_visibility = ["//visibility:public"])
+load("@build_bazel_rules_nodejs//internal/npm_install:node_module_library.bzl", "node_module_library")
+
+filegroup(
+    name = "${pkg}",
+    srcs = [] # we dont care about the actual content, we want a list of dependencies to modify the pnp file around
+)`;
+
+  writeFileSync(`${pkg}/BUILD.bazel`, contents);
+}
+
+async function main() {
+  const packageJson = require(path.join(workspacePath, "package.json"));
```

is a workspace always a package by definition? If yes, is it worth providing a better error message here? If not, how do we handle this?

---

Yes, the workspacePath is the directory containing the package.json file. I suppose it's possible it doesn't exist, so we can return a better error message